### DISCUSSION TITLE
No suggestions in autocomplete

### DISCUSSION THREAD

**Discussion by Serhioromano** [eclipse-langium/langium#777] [Q&A] [answered]
I have managed to implement langium into my VS Code extension. Now I struggle with `.langium` syntax. Example.

```
entry Document:
    (functions+=Function)*;

Function:
    'FUNCTION' name=ID ':' return_type=Dtypes 'END_FUNCTION';

Dtypes returns string:
    'BOOL' | 'SINT' | 'USINT' | 'INT' | 'UINT' | 'DINT' | 'UDINT' | 'LINT' | 'ULINT' | 'REAL'
    | 'LREAL' | 'TIME' | 'TOD' | 'DT' | 'BYTE' | 'WORD' | 'DWORD';

terminal ID: /[_a-zA-Z][\w_]*/;
```

This is simplified version of file. But when I type in document 

```
FUNCTION MyName : 
```

When I start type BOOL it immediately appear but other not. Why?
```

**Comment by msujew** [comment, marked as answer]
Hey @Serhioromano,

Interesting, it seems like the malformed input (i.e. 'B' which doesn't match any of the tokens) breaks the `findLeafNodeAtOffset` method and makes it return undefined. That will then result in no completion results:

https://github.com/langium/langium/blob/a18ebca4ce58391d1d7bbb7dc1509b6ec597bb23/packages/langium/src/lsp/completion/completion-provider.ts#L73-L79

I've created https://github.com/langium/langium/issues/778 to track this.
